### ISSUE TITLE
Merging categoricals does not set Invalid correctly for missing values

### ISSUE THREAD

**Issue by MarcMassar** [rtosholdings/riptable#135]
When doing a merge, 'missing' values (e.g. right side column values in rows that are 'left_only' in a left merge) get an nan/invalid value. But if the merged column is a categorical the invalid data is not set correctly, so isnan() does not work correctly. 

```
lhs = rt.Dataset({'sym': ['AAPL', 'GE']})

rhs = rt.Dataset({'sym': ['AAPL'],
                  'c': rt.Cat(['up']),
                  'numeric': [1.]})

merged = rt.merge2(lhs, rhs, on=['sym'])


#   sym    c         numeric
-   ----   -------   -------
0   AAPL   up           1.00
1   GE     !<-128>       nan         <----------- the numeric column has a nan, but the cat is ???


print(f'{merged.c.isnan() = }')
merged.c.isnan() = FastArray([False, False])    <--------------  should be [False, True]


merged.c is:
Categorical([up, !<-128>]) Length: 2
  FastArray([   1, -128], dtype=int8) Base Index: 1
  FastArray([b'up'], dtype='|S2') Unique count: 1        <------ I guess this is missing an 'Inv' category, 
                                                                 and merged.ca.invalid_category should be set to that.

```

**Comment by jack-pappas** [comment]
@MarcMassar What version of riptable are you using? This looks like the bug that was fixed in v1.0.43: https://github.com/rtosholdings/riptable/releases/tag/v1.0.43

**Comment by MarcMassar** [comment]
You are right, my bad. I was on 1.0.42 and I can't reproduce the issue when I upgrade to 1.0.45

This can be closed.
